libs/datamodel/connectors/sql-datamodel-connector/src/mysql_datamodel_connector.rs

```diff
@@ -137,6 +138,23 @@ impl Connector for MySqlDatamodelConnector {
         &self.capabilities
     }
 
+    fn validate_field(&self, field: &Field) -> Result<(), ConnectorError> {
+        if let FieldType::NativeType(_scalar_type, native_type) = field.field_type() {
+            match native_type.name.as_str() {
+                TEXT_TYPE_NAME => {
+                    if field.is_unique() {
+                        return Err(ConnectorError::new_incompatible_native_type_with_unique(
+                            TEXT_TYPE_NAME,
+                            "MySQL",
+                        ));
+                    }
+                }
+                _ => {}
+            }
```

Comment:
Looks really neat and simple. It would probably be even more concise if you would use an `if` instead of a `match`.